lib/scope.js

```diff
@@ -2503,6 +2514,29 @@ module.exports = {
       await scope.detectDownloadComplete( scope );
     },  // Ends scope.steps.download()
 
+    compare_pdfs: async function (scope, {existing_pdf_path, new_pdf_path}) {
+      let existing_paths = await scope.findFiles(scope, {to_find_names: [existing_pdf_path]});
+      if (existing_paths.length == 0) {
+        // The row wasn't used
+        let msg = `Could not find the existing PDF at ${ existing_pdf_path }`;
+        expect(existing_paths.length, msg).to.be.above(0);
```

Comment:
Good message. We should probably add that to the report too, though. Untested:

```suggestion
        let msg = `Could not find the existing PDF at ${ existing_pdf_path }`;
        reports.addToReport(scope, { type: `error`, value: msg });
        expect(existing_paths.length, msg).to.be.above(0);
```

Just thinking over this - In the past, we've decided to err towards being permissive and warning users as opposed to erroring, saying something like "so we couldn't compare PDFs". That said, you might be right to error here. If we just warn and the tests pass, that seems misleading.

For far in the future: I wonder if there's a way to have both - remember this, but also finish the test, and error later, like the a11y stuff.

Reply:
Just as a note, #687 is about to add [a warning](https://github.com/SuffolkLITLab/ALKiln/pull/755/files#diff-4b632aedc4c91285de358dd7cbcc91a52724745c5c759b81b3a6d80e8d6d7abcR1824) about not finding files. When that's split into `find_files`, that might be a better place to put a warning. We can then error here. We can probably keep this and choose later where is a more appropriate to put it and what it should look like.

Reply:
Sorry, forgot about adding to the report. IMO, yeah, a warning in `findFiles`, and but potentially error in the function that calls `findFiles` is a good option.

TBH doing what we do with the accessibility stuff is a good idea, and not hard to implement. It will add to #748, but whatever solution we have for that would also fix this as well. I'll try it out, if it's taking too long, I'll make an issue for it.

Reply:
Found a good solution for this, I think it works out really well, except for the above mentioned extra screenshot with #748, but this and the accessibility code is very closely related, and the solution to one should fix the other.